tests/test_types.py

```diff
@@ -2540,3 +2541,34 @@ class Model(BaseModel):
         v: Deque[int]
 
     assert Model(v=deque((1, 2, 3))).json() == '{"v": [1, 2, 3]}'
+
+
+@pytest.mark.parametrize('value_type', (None, NoneType, NoneLiteral))
```

Comment:
I'd add a fresh `Literal[None]` here, to check that it works even when the annotation is a different object to the `NoneLiteral` constant.

Reply:
Thanks for reviewing! https://github.com/samuelcolvin/pydantic/pull/2149/commits/832068d322ad384def00681f6d26d6225e283e29